nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java

```diff
@@ -213,6 +213,146 @@ public void nullableAssignmentParameterArray() {
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "import java.util.List;",
+            "class Test {",
+            "  class Super {",
+            "    @Nullable Integer[] foo() { return new @Nullable Integer[0]; }",
+            "    Integer[] bar() { return new Integer[0]; }",
+            "  }",
+            "  class Sub extends Super {",
+            "    @Override",
+            "    Integer[] foo() { return new Integer[0]; }",
+            "    @Override",
+            "    // BUG: Diagnostic contains: Method returns @Nullable Integer[], but overridden method returns Integer[]",
+            "    @Nullable Integer[] bar() { return new @Nullable Integer[0]; }",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void overridesParameterType() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "import java.util.List;",
+            "class Test {",
+            "  class Super {",
+            "    void foo(@Nullable Integer[] p) { }",
+            "    void bar(Integer[] p) { }",
+            "  }",
+            "  class Sub extends Super {",
+            "    @Override",
+            "    // BUG: Diagnostic contains: Parameter has type Integer[], but overridden method has parameter type @Nullable Integer[]",
+            "    void foo(Integer[] p) { }",
+            "    @Override",
+            "    void bar(@Nullable Integer[] p) { }",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void ternaryOperator() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static Integer[] testPositive(Integer[] p, boolean t) {",
+            "    // BUG: Diagnostic contains: Conditional expression must have type Integer[]",
+            "    Integer[] t1 = t ? new Integer[0] : new @Nullable Integer[0];",
+            "    // BUG: Diagnostic contains: Conditional expression must have type",
+            "    return t ? new @Nullable Integer[0] : new @Nullable Integer[0];",
+            "  }",
+            "  static void testPositiveTernaryMethodArgument(boolean t) {",
+            "    // BUG: Diagnostic contains: Conditional expression must have type",
+            "    Integer[] a = testPositive(t ? new Integer[0] : new @Nullable Integer[0], t);",
+            "  }",
+            "  static @Nullable Integer[] testNegative(boolean n) {",
+            "    @Nullable Integer[] t1 = n ? new @Nullable Integer[0] : new @Nullable Integer[0];",
+            "    @Nullable Integer[] t2 = n ? new Integer[0] : new @Nullable Integer[0];",
+            "    return n ? new @Nullable Integer[0] : new @Nullable Integer[0];",
+            "  }",
```

Comment:
test negative arg case missing (mostly for the sake of symmetry 🙂 )

Reply:
075fa41ae62078cb740610a247184688e3030d18